rdtools/degradation.py

```diff
@@ -80,3 +80,293 @@ def degradation_with_ols(normalized_energy):
+    degradation_values = {
+        'Rd_median': med1,
+        #'slope': m_cd,
+        #'intercept': b_cd,
+        #'rmse': rmse,
+        #'slope_stderr': stderr_m,
+        #'intercept_stderr': stderr_b,
+        'Rd_stderr_pct': unc1,
+        'YoY_filtered':YoY_filtered1
+        }
+    return(degradation_values)
+      
+    # End YOY approach
+    ################################################################################################################################
+
+      
+def degradation_ARIMA(normalized_energy):
+    # (Rd_decomp,SE_Rd_decomp,test_trend,p,df4) = classical_decomposition(normalized_energy,Month)
+    #
+    # ARIMA seasonal decomponsition method using statsmodel.api
+    # from Dirk Jordan (NREL) 12/10/16
+    # input parameters:  
+    #       dataframe['normalized_energy'] : corrected performance ratio.
+    #output parameters:
+    #   (Rd_decomp,SE_Rd_decomp,test_trend,p) : tuple of annual degradation rate and standard error
+    # TODO:  check timebase for other input time options (day, week)    
+
+    ################################################################################################################################
+    # ARIMA approach - Seasonal decompostion is a special type of ARIMA model   
+
+    # deal with missing values
+    #y = y.interpolate(inplace=True)
+    
+    res = sm.tsa.seasonal_decompose(normalized_energy, freq=12)
+    #resplot = res.plot()         
+    y_decomp = res.trend.dropna()        
+    df4 = pd.DataFrame({'MS':y_decomp.index,'normalized_energy':y_decomp.values})
+    df4['Month'] = range(0, len(df4))
+    y3=df4['normalized_energy']
+    x3=df4['Month']           
+    
+    
+    
+    if len(df4) <=2:
+        Rd_decomp = 11111
+        SE_Rd_decomp = 11111
+        test_trend = 11111
+        p = 11111
+        print '\nNot enough data for ARIMA decomposition:'
+       
+    else:
+ 
+       # OLS regression with constant
+        x3=sm.add_constant(x3)
+        model3 = sm.OLS(y3,x3).fit()
+    
+        b_decomp = model3.params['const']
+        m_decomp = model3.params['Month']
+        SE_b_decomp = model3.bse['const']
+        SE_m_decomp = model3.bse['Month']
+    
+        Rd_decomp = (m_decomp * 12) / b_decomp * 100
+        SE_Rd_decomp = np.sqrt(np.power(SE_m_decomp * 12/b_decomp, 2) + np.power((-12*m_decomp/b_decomp**2) * SE_b_decomp, 2))*100
+
+        #print '\nDegradation and Standard Error of ARIMA decomposition:'
+        #print 'Rd = {:.2f} +/- {:.2f}'.format(Rd_decomp, SE_Rd_decomp)              
+       
+        test_trend,h,p,z = _mk_test(y_decomp,alpha=0.05)  
+        #print test_trend, h, p    
+    
+        
+        degradation_values = {
+        'Rd_pct': Rd_decomp,
+        'test_trend': test_trend,
+        'p': p,
+        #'rmse': rmse,
+        #'slope_stderr': stderr_m,
+        #'intercept_stderr': stderr_b,
+        'Rd_stderr_pct': SE_Rd_decomp,
+        'Dataframe':df4,
+        }
+    return(degradation_values)
```

Comment:
`degradation_values` is not defined in the `if len(df4) <=2` if-block, therefore this return statement will throw an exception in that case.